### PR TITLE
Bug fix/copy schmutz

### PR DESCRIPTION
**Pull request by dothebart**
This patch overcomes the caching issues that we have on MSVC + MSBuild.

### CODE REVIEW

**Comment by dothebart** [comment]
Tests green.

File: 3rdParty/iresearch/core/CMakeLists.txt [hunk 1]
```diff
@@ -83,6 +83,14 @@ if (USE_OPTIMIZE_FOR_ARCHITECTURE)
   )
 endif()
 
+if (GENERATED_PARSER_SOURCES)
```
Comment (vasiliy-arangodb) [inline 3rdParty/iresearch/core/CMakeLists.txt]: the IResearch library must ensure that all its dependencies are built present, the first logic branch breaks this precondition

File: 3rdParty/CMakeLists.txt [hunk 2]
```diff
@@ -180,24 +185,6 @@ if (USE_IRESEARCH)
     "${IRESEARCH_ROOT}/cmake" # to find iResearch dependencies
   )
 
-  # cmake will seemingly randomly delete files from ${IRESEARCH_BUILD_DIR} during configuration
-  # so must copy required files only after the cmake configuration phase is complete, not via file(COPY ...)
-  # this command must _always_ be executed because the subsequent cmake configuration runs will once again delete files
```
Comment (vasiliy-arangodb) [inline 3rdParty/CMakeLists.txt]: The cmake build process seems to sometimes remove files required for the IResearch 3rd-party library build process
therefore this workaround is in-place to ensure that the required files are always present when IResearch is built

Comment (dothebart) [reply 3rdParty/CMakeLists.txt]: This patch will not copy files into the build directory at all. It simply will overrule their source file and the generation mechanism. 
Please provide a way to reproduce your claim that this doesn't work as intended.

File: 3rdParty/iresearch/core/CMakeLists.txt (hunk 1 above)
Comment (dothebart) [reply 3rdParty/iresearch/core/CMakeLists.txt]: The invoker overrules generation by specifying where to get the pre-generated file.

Comment (vasiliy-arangodb) [reply 3rdParty/iresearch/core/CMakeLists.txt]: this is not part of the scope of responsibility of the IResearch library, please provide a clean solution that does not cross scope of responsibility boundaries

File: 3rdParty/CMakeLists.txt (hunk 2 above)
Comment (vasiliy-arangodb) [reply 3rdParty/CMakeLists.txt]:
```
> Please provide a way to reproduce your claim that this doesn't work as intended.
remove the copying of files and try to rebuild, every so often, seemingly randomly, the build will fail due to these files missing from the directory
```
once the root cause of this issue is addressed then it is totally reasonable to remove file copying during the build stage, and only leave it for the configure stage

**Pull request merged**